### PR TITLE
Fix grpcgen TestMtls

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Fix https://github.com/istio/istio/issues/36804

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/fake.go [hunk 1]
```diff
@@ -172,7 +172,6 @@ func NewConfigGenTest(t test.Failer, opts TestOptions) *ConfigGenTest {
 		if err := env.InitNetworksManager(&FakeXdsUpdater{}); err != nil {
 			t.Fatal(err)
 		}
-		env.PushContext = model.NewPushContext()
 		if err := env.PushContext.InitContext(env, nil, nil); err != nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/fake.go:175]: SkipRun is very error prone, @howardjohn Maybe we should remove this flag

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
Still a few flakes: https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/37112/unit-tests_istio/1488922988769185792. Need to look into it

**Pull request merged**